src/header/types/accept.rs

```diff
@@ -0,0 +1,325 @@
+    /// use mime::*;
+    /// use std::str::FromStr;
+    /// 
+    /// let typed_header = vec![MediaType::new(mime::STAR_STAR, None), MediaType::new(Mime::from_str("video/*").unwrap(), Some(QualityParam::new(0.5)))]
+    ///     .into_iter()
+    ///     .collect::<Accept>();
+    /// let expected_raw_headers = vec![
+    ///     vec![HeaderValue::try_from("*/*, video/* ;q=0.5").unwrap()],
+    ///     vec![HeaderValue::try_from("video/* ;q=0.5, */*").unwrap()],
+    /// ];
+    /// let mut raw_header = vec![];
+    /// typed_header.encode(&mut raw_header);
+    /// assert!(raw_header == expected_raw_headers[0] ||
+    ///         raw_header == expected_raw_headers[1]);
+    /// ```
+    fn encode<Target>(&self, values: &mut Target)
+    where
+        Target: Extend<HeaderValue>
+    {
+        // Unsafe Justification
+        //
+        // Header values must be valid UTF-8, and since we know that the [`RangeFormat`] type
+        // guarantees valid ASCII-US (with no newlines), it satisfies the constraints.
+        let value = self.iter().map(MediaType::as_str).join(", ");
+        values.extend(once(unsafe { HeaderValue::from_string_unchecked(value) }));    }
+
+    fn header_name() -> &'static HeaderName
+    where   
+        Self: Sized
+    {
+        &HeaderName::Accept
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MediaType{
+    m_type: Mime,
+    quality: Option<QualityParam>
+
+}
+
+impl Default for MediaType {
+
+    fn default() -> Self {
+        MediaType{
+            m_type: mime::TEXT_PLAIN,
+            quality: None
+        }
+    }
+}
+
+impl Eq for MediaType {}
+
+impl Hash for MediaType {
+
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.m_type.hash(state);
+    }
+}
+
+impl MediaType{
+
+    pub fn new(m_type: Mime, quality: Option<QualityParam>) -> Self {
+        MediaType{
+            m_type,
+            quality
+        }
+    }
+
+    /// Returns a `&str` representation of the media type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::convert::TryFrom;
+    ///
+    /// use rtsp::header::types::accept::MediaType;
+    /// use rtsp::header::types::accept::QualityParam;
+    /// use mime::*;
+    /// use std::str::FromStr;
+    ///
+    /// assert_eq!(MediaType::new(mime::STAR_STAR, None).as_str(), "*/*");
+    /// assert_eq!(MediaType::new(Mime::from_str("video/*").unwrap(), Some(QualityParam::new(0.5))).as_str(), "video/* ;q=0.5");
+    /// ```
+    pub fn as_str(&self) -> String {
+        let m_type = self.m_type.to_string();
+        match &self.quality {
+            Some(quality) => format!("{} ;{}", m_type, quality.as_str()),
+            None => format!("{}", m_type)
+        }
+    }
+}
+
+impl<'accept> TryFrom<&'accept [u8]> for MediaType {
+    type Error = AcceptError;
+
+    fn try_from(value: &'accept [u8]) -> Result<Self, Self::Error> {
+        let mut split = value.split(|element| element.clone() == ';' as u8);
```

Comment:
`element.clone() == ';' as u8` can be simplified to `|&element| element == b';'` Also, since you aren't considering other parameters here, use `splitn` with 2 instead